rpmio/rpmkeyring.c

```diff
@@ -153,37 +155,85 @@ rpmPubkey rpmPubkeyNew(const uint8_t *pkt, size_t pktlen)
     return key;
 }
 
-static rpmPubkey rpmPubkeyNewSubkey(pgpDigParams pgpkey)
+static rpmPubkey rpmPubkeyNewSubkey(rpmPubkey primarykey, pgpDigParams pgpkey)
 {
     rpmPubkey key = new rpmPubkey_s {};
-    /* Packets with all subkeys already stored in main key */
+    /* Packets with all subkeys already stored in primary key */
+    key->primarykey = primarykey;
```

Comment:
I suppose it should actually take the reference here already, might simplify the rest of stuff as well.
But l agree, lets take another look at the whole in #3350/#3351, it should be easier to see the big picture when this is already in place. We're both getting battle-weary on this PR.